### ISSUE TITLE
ImageTileSource bug

### ISSUE THREAD

**Issue by pearcetm** [openseadragon/openseadragon#2159]
In the ImageTileSource class, the `_freeupCanvasMemory()` function iterates over levels and sets canvas width and height to zero:

```
_freeupCanvasMemory: function () {
            for (var i = 0; i < this.levels.length; i++) {
                this.levels[i].context2D.canvas.height = 0;
                this.levels[i].context2D.canvas.width = 0;
            }
        },
```

The problem is, the `context2D` property does not always exist - see the `_buildLevels()` code.

```
_buildLevels: function () {
            var levels = [{
                    url: this._image.src,
                    width: this._image.naturalWidth,
                    height:  this._image.naturalHeight
                }];

            if (!this.buildPyramid || !$.supportsCanvas || !this.useCanvas) {
                // We don't need the image anymore. Allows it to be GC.
                delete this._image;
                return levels; //<------ context2d is never added to the object
            }
            ...
    }
```  

This causes the viewer to stop working when attempting to close an ImageTileSource that does not use `buildPyramid=true`

I've fixed this in my local code by changing the _freeupCanvasMemory function to check for the existence of context2D before attempting to access the canvas. I don't really use this type of tile source very often so I'm not deeply familiar with the workings of the class, and there may be other problems related to this that also need to be addressed. I also don't know if my fix is "optimal" in the sense of adding a conditional check within each level rather than once outside of the loop, etc. Code snippet below that fixes the immediate problem:

```
_freeupCanvasMemory: function () {
            for (var i = 0; i < this.levels.length; i++) {
                if(this.levels[i].context2D){
                    this.levels[i].context2D.canvas.height = 0;
                    this.levels[i].context2D.canvas.width = 0;
                }
            }
        },
```

**Comment by iangilman** [comment]
Good catch! That seems like a fine fix. I don't know that you could optimize it further, since each level has its own context2D. It seems like a pretty safe fix as well. 

Would you be up for making a pull request with this change?

**Comment by pearcetm** [comment]
I created a pull request (https://github.com/openseadragon/openseadragon/pull/2162). It's actually my first time doing this... I'm not sure how to link this issue thread with the PR - if you can help out I'd appreciate it!
